rust/src/lib.rs

```diff
@@ -2142,24 +2162,27 @@ pub unsafe extern "C" fn zcashlc_propose_shielding(
                     })
             })?;
 
-        let from_addrs = match transparent_receiver.map_or_else(||
-            if account_receivers.len() > 1 {
-                Err(anyhow!(
-                    "Account has more than one transparent receiver with funds to shield; this is not yet supported by the SDK. Provide a specific transparent receiver to shield funds from."
-                ))
-            } else {
-                Ok(account_receivers.iter().next().map(|(a, v)| (*a, *v)))
-            },
-            |addr| Ok(account_receivers.get(&addr).map(|value| (addr, *value)))
-        )?.filter(|(_, value)| *value >= shielding_threshold) { Some((addr, _)) => {
-            [addr]
-        } _ => {
-            // There are no transparent funds to shield; don't create a proposal.
+        // If a specific receiver is specified, select only value for that receiver; otherwise,
+        // select value for all receivers. See the warnings associated with the documentation
+        // of the `transparent_receiver` argument in the method documentation for privacy
+        // considerations.
```

Comment:
As a general rule I prefer that if a semantic change could potentially reduce security for existing callers (in any situation), it is deliberately API/ABI-breaking. In this particular case the Swift SDK is basically the only user of the API so I'm not going to insist on it.